packages/payment-processor/src/payment/utils.ts

```diff
@@ -161,6 +184,44 @@ export function validateErc20FeeProxyRequest(
   }
 }
 
+/**
+ * Validates the parameters for an ERC20 Fee Proxy payment.
+ * @param request to validate
+ * @param tokenAddress token address to pay with
+ * @param amount optionally, the custom amount to pay
+ * @param feeAmountOverride optionally, the custom fee amount
+ */
+export function validateConversionFeeProxyRequest(
+  request: ClientTypes.IRequestData,
+  path: string[],
+  amount?: BigNumberish,
+  feeAmountOverride?: BigNumberish,
+): void {
+  const { feeAddress, feeAmount, tokensAccepted } = getRequestPaymentValues(request);
+  const amountToPay = getAmountToPay(request, amount);
+  const feeToPay = bigNumberify(feeAmountOverride || feeAmount || 0);
+
+  if (!!feeAmount !== !!feeAddress) {
+    throw new Error('Both fee address and fee amount have to be declared, or both left empty');
+  }
+  if (amountToPay.isZero() && feeToPay.isZero()) {
+    throw new Error('Request payment amount and fee are 0');
+  }
```

Comment:
Small factorization here I think with something like `validateFeesAndAmount(request)` used for fee-proxy and any-to-erc20 ?